plugin-best-practices-plugin/src/main/kotlin/com/autonomousapps/internal/analysis/IssueListener.kt

```diff
@@ -316,3 +317,25 @@ internal class GetProjectListener : AbstractIssueListener() {
     return MethodNode.Metadata(isTaskAction)
   }
 }
+
+/** Invokes Eager APIs instead of Lazy ones.
+ * @see <a href="https://docs.gradle.org/current/userguide/lazy_configuration.html">Lazy Configuration</a>
+ * @see <a href="https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:old_vs_new_configuration_api_overview">Old vs New API Overview</a>
+ */
+internal class EagerApisListener : AbstractIssueListener() {
+
+  private val eagerApis = mapOf(
```

Comment:
My thinking for this structure is to have a `Map<K,V>`, where `K` is the class name, and `V` is a `Set` of eager methods that can be accessed through it. Currently, this set only contains 3 commonly used methods, but there many more that can be added later as well. 

I chose this because it would be read-performant for this particular implementation, but I have concerns about maintainability going forward as this Map may require some upkeep. Does anyone have better ideas?

Reply:
I don't have any problem with this. I don't think performance is a major concern here tbh.